src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -520,6 +520,12 @@ public synchronized void setCurrentVote(Vote v){
     protected boolean quorumListenOnAllIPs = false;
 
     /**
+     * Keeps time taken for leader election in milliseconds. Sets the value to
+     * this variable only after the completion of leader election.
+     */
+    private long electionTimeTaken = -1;
```

Comment:
sincere question: does it make sense to make this field `volatile`? I know there are some concurrency guarantees but not sure if it's worth change it.